lib/async_hooks.js

```diff
@@ -516,3 +517,10 @@ Object.defineProperty(module.exports, 'triggerId', {
   }, 'async_hooks.triggerId is deprecated. ' +
      'Use async_hooks.triggerAsyncId instead.', 'DEP0071')
 });
+
+// Integration with V8 Inspector AsyncTask* API
+// FIXME(bajtos) move this code to src/inspector_agent.cc
+// Agent::Start() should enable the hook, Agent::Stop() should disable it
```

Comment:
Agreed.

I have been looking into this issue in the past days, the problem is more complex than I initially thought. Typically `Agent::Start` is called directly from `node::Start`, long time before `bootstrap_node.js` is loaded. When debugging via `--inspect-brk`, `bootstrap_node.js` is not started until a debugger is attached.

One option is to enable the hook at the time when a client connects (in `Agent::Connect`), plus defer hook enablement until `bootstrap_node.js` is called in case of `--inspect-brk`, because in case of `--inspect-brk`, the client connects before we can create the inspector async hook. However, I find the user experience of this solution rather poor. Because async stack trace tracing is enabled only at the time the debugger connects, any async transitions created at startup time (e.g. like setting up an HTTP server, creating timeouts, etc.) are lost.

Here is my current plan:
 - Somewhere in `src/node.cc`, set a flag that can be used by `lib/internal/inspector_async_hook` to decide whether the hook should be enabled at Node.js start/boot time. This should nicely handle both `--inspect` and `--inspect-brk` cases, as long as Node.js bootstrapping process stays synchronous.
 - At the same time, in order to support `process._startDebug`, I'll still need to pass the "enable" function into C++ land, so that we can enable the hook at runtime when the debugger was enabled via a signal (`_startDebug`).

I estimate I'll need few more days to finish this, and since I am on vacation next week, I may not be able to push the update until the week of July 10th.